cpp/cppsdk/gsdk.cpp

```diff
@@ -289,12 +289,14 @@ namespace Microsoft
                 try {
                     if (heartbeatResponse.isMember("sessionConfig"))
                     {
+						std::unordered_map<std::string, std::string> tempConfig = std::unordered_map<std::string, std::string>(m_configSettings);
```

Comment:
>std::unordered_map<std::string, std::string>(m_configSettings) [](start = 64, length = 62)

This could just be m_configSettings.

Reply:
Agreed. Here we want to run the copy-constructor. I'm still a bit concerned about us being in the middle of copying m_configSettings in one call to decodeHeartbeatResponse when another call assigns a new map to m_configSettings, causing the previous one to be deleted...


---
In reply to: [377802573](https://github.com/PlayFab/gsdk/pull/43#discussion_r377802573) [](ancestors = 377802573)